docs/src/tutorials/symbmc.md

```diff
@@ -0,0 +1,690 @@
+# Symbolic Model Checking
+
+This brief introduction to symbolic model checking discusses the following:
+
+  1. State-spaces and transition systems
+  1. What is a symbolic state?
+  1. What are symbolic traces?
+  1. How do I interpret Apalache counterexamples?
+
+A glossary of notation and definitions can be found [below](#notation-and-definitions)
+
+## On state-spaces and transition systems
+A TLA+ specification defines a triple \\((S,S_0,\to)\\), called a _transition system_. \\(S\\) is the _state space_, \\(S_0\\) is the set of initial states \\(\left(S_0 \subseteq S\right)\\), and \\(\to\\) is the transition relation, a subset of \\(S^2\\).
+
+### State spaces
+The structure of a single state depends on the number of variables a specification declares. For example, if a specification declares 
+```tla
+VARIABLE A1, A2, A3, ..., Ak
+```
+
+then a _state_ is a mapping \\([A_1 \mapsto a_1, \dots, A_k \mapsto a_k]\\), where \\(a_i\\) represents the value of the variable `Ai`, for each \\(i = 1,\dots,k\\). Here, we represent TLA+ variable names as unique formal symbols, where, for example the TLA+ variable `A1` is represented by the formal symbol \\(A_1\\). By convention, we will use markdown-syntax to refer to objects in TLA+ specifications, and latex notation otherwise.
+The state space \\(S\\) is then the set of all such mappings, i.e. the set of all possible combinations of values that variables may hold.
+For brevity, whenever the specification defines exactly one variable, we will treat a state as a single value \\(a_1\\) instead of the mapping \\([A_1 \mapsto a_1]\\).
+
+In untyped TLA+, one can think of \\(S\\) as \\(U^{\\{A_1,\dots, A_k\\}}\\), that is, the set of all mappings, which assign a value in \\(U\\), the universe of all TLA+ values, to each symbol. This set is naturally isomorphic to \\(U^k\\). In typed TLA+, such as in Apalache, where variable declarations look like:
+```tla
+VARIABLE 
+  \* @type: T1;
+  A1, 
+  ..., 
+  \* @type: Tk;
+  Ak
+```
+
+ \\(S\\) is additionally restricted, such that for all \\(s \in S\\) each symbol \\(A_i\\) maps to a value \\(s(A_i) \in V_i\\), where \\(Vi \subset U\\) is the set of all values, which hold the type \\(T_i\\), for each \\(i = 1,\dots,k\\).
+For example, in the specification with
+```tla
+VARIABLE 
+  \* @type: Bool;
+  a, 
+  \* @type: Bool;
+  b
+```
+
+The state space is \\(\mathbb{B}^{\\{x,y\\}}\\) when considering types, since each variable can hold one of two boolean values. In the untyped setting, the state space is infinite, and contains states where, for example, \\(a\\) maps to `[z \in 1..5 |-> "a"]` and \\(b\\) maps to  `CHOOSE p \in {}: TRUE`.
```

Comment:
```suggestion
The state space is \\(\mathbb{B}^{\\{x,y\\}}\\) when considering types, since each variable can hold one of two boolean values. In the untyped setting, the state space is infinite, and contains states where, for example, \\(a\\) maps to `[z \in 1..5 |-> "a"]` and \\(b\\) maps to  `CHOOSE p: TRUE`.
```